File: fix_settings_usages.js
const fs = require('fs');
const glob = require('glob'); // Not available? I'll just use a recursive walk.
const path = require('path');

function walk(dir) {
    let results = [];
    let list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = dir + '/' + file;
        let stat = fs.statSync(file);
        if (stat && stat.isDirectory()) { 
            results = results.concat(walk(file));
        } else { 
            if (file.endsWith('.tsx') || file.endsWith('.ts')) {
                results.push(file);
            }
        }
    });
    return results;
}

let files = walk('packages/ui-shared/src');

for (let file of files) {
    let content = fs.readFileSync(file, 'utf8');
    let original = content;

    content = content.replace(/useChordStore\(\s*\(\s*s\s*\)\s*=>\s*s\.settings\s*\)/g, 'useSettingsStore((s) => s.settings)');
    content = content.replace(/useChordStore\(\s*\(\s*state\s*\)\s*=>\s*state\.settings\s*\)/g, 'useSettingsStore((state) => state.settings)');
    
    content = content.replace(/useChordStore\(\s*\(\s*s\s*\)\s*=>\s*s\.lastSession\s*\)/g, 'useSettingsStore((s) => s.lastSession)');
    content = content.replace(/useChordStore\(\s*\(\s*state\s*\)\s*=>\s*state\.lastSession\s*\)/g, 'useSettingsStore((state) => state.lastSession)');

    content = content.replace(/useChordStore\.getState\(\)\.settings/g, 'useSettingsStore.getState().settings');
    content = content.replace(/useChordStore\.getState\(\)\.settingsController/g, 'useSettingsStore.getState().settingsController');

    // Missing settingsController imports or definitions
    content = content.replace(/const\s+settingsController\s*=\s*useSettingsStore\.getState\(\)\.settingsController/g, 'const settingsController = useSettingsStore.getState().settingsController');

    // Missing updateSettings
    content = content.replace(/updateSettings/g, 'useSettingsStore.getState().settingsController.updateSettings');
    // Wait, updateSettings is too generic.
    
    if (content !== original) {
        // Add import if missing
        if (!content.includes('useSettingsStore')) {
            content = content.replace(/import\s*\{([^}]*)\}\s*from\s*'@workspace\/studio-core';/, "import { $1, useSettingsStore } from '@workspace/studio-core';");
        }
        fs.writeFileSync(file, content, 'utf8');
    }
}

File: fix_settings_usages.py
import re
import os

files = [
    'src/lib/diagnostics/activityLogger.ts',
    'src/lib/diagnostics/devTools.ts',
    'src/lib/hooks/useT.ts',
    'src/lib/services/permissions.ts',
    'src/lib/startup/startupCoordinator.ts',
    'src/lib/sync/sync.ts',
    'src/lib/updater/telemetry.ts',
    'src/repositories/UserRepository.ts'
]

for file in files:
    path = os.path.join('packages', 'studio-core', file)
    if os.path.exists(path):
        with open(path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Replace useChordStore.getState().settings with useSettingsStore.getState().settings
        # And ensure useSettingsStore is imported
        if 'useChordStore.getState().settings' in content:
            content = content.replace('useChordStore.getState().settings', 'useSettingsStore.getState().settings')
            if 'useSettingsStore' not in content:
                content = "import { useSettingsStore } from '../../store/useSettingsStore';\n" + content
                content = content.replace("import { useSettingsStore } from '../../store/useSettingsStore';\nimport { useSettingsStore }", "import { useSettingsStore }") # basic deduplication just in case
                
        with open(path, 'w', encoding='utf-8') as f:
            f.write(content)
